manifests/noderesourcetopology/deploy.yaml

```diff
@@ -27,25 +27,17 @@ spec:
           imagePullPolicy: Never
           command:
           - /bin/kube-scheduler
-          - --scheduler-name=topo-aware-scheduler
-          - --authentication-kubeconfig=/etc/kubernetes/scheduler.conf
-          - --authorization-kubeconfig=/etc/kubernetes/scheduler.conf
```

Comment:
I think these two args are still needed ([doc](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/)). They're neither deprecated nor derive the settings from scheduler CC:

```
kubeconfig  file pointing at the 'core' kubernetes server with enough rights to  create tokenreviews.authentication.k8s.io. This is optional. If empty,  all token requests are considered to be anonymous and no client CA is  looked up in the cluster.
```

Reply:
The doc says it's optional, so those settings must be accessible from somewhere else too right? Are they available if you just use the `--kubeconfig` flag?

Reply:
But in the documentation I see using `--kubeconfig` flag is deprecated?
```
--kubeconfig string
DEPRECATED: path to kubeconfig file with authorization and master location information. This parameter is ignored if a config file is specified in --config.
```
and we are have worked to avoid accessing the kubeconfig in the favour of obtaining it from the scheduler framework handler.  I am a little confused here to be honest.

Reply:
> we still need to mount in the kubeconfig into the scheduler plugin deployment? I thought that kubernetes/kubernetes#100644 was proposed to overcome the need to specify the kubeconfig path in the plugin arguments?

kubernetes/kubernetes#100644 does remove the need to specify the kubeconfig path in the plugin arguments. These flags aren't plugin arguments, they're core to the scheduler itself

Reply:
> @Huang-Wei If we still need `--authentication-kubeconfig` and `--authorization-kubeconfig` then it means we still need to mount in the kubeconfig into the scheduler plugin deployment?

That's correct, just think of scheduler-plugins as a variant of the default scheduler. You need to at least provide a kubeConfig :-)

Reply:
What kubernetes/kubernetes#100644 tries to solve is in the Go SDK's perspective, the out-of-tree plugins don't need to specify the kubeConfig again.

Reply:
Okay, so why don't we have it specified here: https://github.com/kubernetes-sigs/scheduler-plugins/blob/6e58632f57f80274dc5e0f97df7f1d1907cc7c25/manifests/install/charts/as-a-second-scheduler/templates/deployment.yaml#L44-L48?

Reply:
Also, we experimented with deployment of scheduler plugin without these flags and it worked fine and did not complain. I wonder why it is working for us if it shouldn't :thinking:

Reply:
> Okay, so why don't we have it specified here?

That's a mistake, we should specify it in that deployment as well.

> Also, we experimented with deployment of scheduler plugin without these flags and it worked fine and did not complain.

As the doc says, it's optional. In rare cases, we can use another kubeconfig to serve the token requests. The hack script (hack/local-up-cluster.sh) in k/k also specifies it in controller-manager and kube-scheduler's startup.